### PR TITLE
feat: update schematics: mechanism for adding parameter to deprecated constructor

### LINKED ISSUE #6432: Implement update schematics example for adding parameter to deprecated constructor
Implement helper method(s) that use AST to locate the constructor and that can do the necessary changes.

### PR DESCRIPTION
**Pull request by znikola**
closes #6432

### CODE REVIEW

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 1]
```diff
@@ -126,33 +140,264 @@ export function defineProperty(
...
+  const firstCallExpression = callExpressions[0];
+  const superKeyword = findNodes(
+    firstCallExpression,
+    ts.SyntaxKind.SuperKeyword
+  );
+  if (superKeyword && superKeyword.length === 0) {
+    throw new SchematicsException('No super() call found.');
+  }
+
+  let toAdd = '';
```
Comment (psmul) [inline projects/schematics/src/shared/utils/file-utils.ts]: Can we change it to a more descriptive name?

File: projects/schematics/src/shared/utils/file-utils.ts [hunk 2]
```diff
@@ -126,33 +140,264 @@ export function defineProperty(
...
+  const constructorParameters = findNodes(
+    constructorNode,
+    ts.SyntaxKind.Parameter
+  );
 
-  if (!parameterListNode) {
-    throw new SchematicsException(
-      `No no parameter list found in ${path}'s constructor.`
-    );
+  let toAdd = '';
```
Comment (psmul) [inline projects/schematics/src/shared/utils/file-utils.ts]: Can we change it to a more descriptive name?

File: projects/schematics/src/shared/utils/file-utils_spec.ts [hunk 3]
```diff
@@ -176,20 +203,313 @@ describe('File utils', () => {
     });
   });
 
+  describe('isCandidateForConstructorDeprecation', async () => {
+    it('should return false if the inheritance condition is not satisfied', () => {
+      const content = `
```
Comment (psmul) [inline projects/schematics/src/shared/utils/file-utils_spec.ts]: Can we go with consts before tests (as in `update-cms-component-state_spec.ts`)? Just for more consistency

File: projects/schematics/src/shared/utils/file-utils_spec.ts [hunk 4]
```diff
@@ -176,20 +203,313 @@ describe('File utils', () => {
...
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the constructor condition is not satisfied', () => {
+      const content = `
```
Comment (psmul) [inline projects/schematics/src/shared/utils/file-utils_spec.ts]: Can we go with consts before tests (as in `update-cms-component-state_spec.ts`)? Just for more consistency

File: projects/schematics/src/shared/utils/file-utils_spec.ts [hunk 5]
```diff
@@ -176,20 +203,313 @@ describe('File utils', () => {
...
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the parameter lengths condition is not satisfied', () => {
+      const content = `
```
Comment (psmul) [inline projects/schematics/src/shared/utils/file-utils_spec.ts]: Can we go with consts before tests (as in `update-cms-component-state_spec.ts`)? Just for more consistency

File: projects/schematics/src/shared/utils/file-utils_spec.ts [hunk 6]
```diff
@@ -176,20 +203,313 @@ describe('File utils', () => {
...
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the parameter order is not satisfied', () => {
+      const content = `
```
Comment (psmul) [inline projects/schematics/src/shared/utils/file-utils_spec.ts]: Can we go with consts before tests (as in `update-cms-component-state_spec.ts`)? Just for more consistency

File: projects/schematics/src/shared/utils/file-utils_spec.ts [hunk 7]
```diff
@@ -176,20 +203,313 @@ describe('File utils', () => {
...
+      expect(
+        isCandidateForConstructorDeprecation(
+          source,
+          USER_ADDRESS_SERVICE,
+          parameterClassTypes
+        )
+      ).toEqual(false);
+    });
+    it('should return false if the super() call does NOT exist', () => {
+      const content = `
```
Comment (psmul) [inline projects/schematics/src/shared/utils/file-utils_spec.ts]: Can we go with consts before tests (as in `update-cms-component-state_spec.ts`)? Just for more consistency

File: projects/schematics/src/shared/utils/file-utils.ts (hunk 1 above)
Comment (znikola) [reply projects/schematics/src/shared/utils/file-utils.ts]: Changed to `toInsert`, let me know if you think it should be more descriptive

File: projects/schematics/src/shared/utils/file-utils.ts (hunk 2 above)
Comment (znikola) [reply projects/schematics/src/shared/utils/file-utils.ts]: The same as above

File: projects/schematics/src/shared/utils/file-utils_spec.ts (hunk 3 above)
Comment (znikola) [reply projects/schematics/src/shared/utils/file-utils_spec.ts]: Done for all comments.

**Pull request merged**